packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts

```diff
@@ -497,6 +497,83 @@ describe('Validation at REST level', () => {
     },
   );
 
+  context('with request body validation options - custom keywords', () => {
+    @model()
+    class ProductWithName {
+      @property()
+      id: number;
+
+      @property({required: true, jsonSchema: {validProductName: true}})
+      name: string;
+
+      constructor(data: Partial<ProductWithName>) {
+        Object.assign(this, data);
+      }
+    }
+
+    class ProductController {
+      @post('/products')
+      async create(
+        @requestBody({required: true}) data: ProductWithName,
+      ): Promise<ProductWithName> {
+        return new ProductWithName({id: 1, name: 'prod-1'});
+      }
+    }
+
+    before(() =>
+      givenAnAppAndAClient(ProductController, {
+        nullable: false,
+        logger: false, // Disable log warning - meta-schema not available
+        compiledSchemaCache: new WeakMap(),
+        $data: true,
+        ajvKeywords: true,
+        ajvErrors: {
+          singleError: true,
+        },
+        keywords: {
+          validProductName: {
+            async: true,
```

Comment:
what would happen if I have `async: false` here?

Reply:
Then the `validateProductName` function cannot be `async`.

Reply:
I added a test case.